tools/cli/commands/build.js

```diff
@@ -831,10 +859,37 @@ async function buildProject( t ) {
 		await fs.writeFile( `${ buildDir }/.gitattributes`, rules, { encoding: 'utf8' } );
 	}
 
+	// Get the project version number from the changelog.md file.
+	const projectVersionNumber = '';
+	try {
+		const rl = rlcreateInterface( {
+			input: createReadStream( `${ t.cwd }/CHANGELOG.md`, {
+				encoding: 'utf8',
+			} ),
+			crlfDelay: Infinity,
+		} );
+
+		// eslint-disable-next-line no-unused-vars
+		rl.on( 'line', line => {
+			// Todo: pull the correct version number from the changelog file.
+		} );
+		await once( rl, 'close' );
+
+		if ( ! projectVersionNumber ) {
+			throw new Error( 'Version number not found.' );
+		}
+	} catch ( err ) {
+		await t.setStatus( 'build failure' );
```

Comment:
The status here will get overwritten on line 327 anyway, no need to bother.

For that matter, you could probably just skip the whole `catch` here, just have line 879 throw the error with the right message directly.